test_helm_install_uninstall.py

```diff
@@ -0,0 +1,138 @@
+
+def test_helm_uninstall():
+    try:
+        node_obj = manager.hpe_list_node_objects()
+        node_list = node_obj.items
+        # Fetching master node ip
+        for item in node_list:
+            if item.spec.taints != "None":
+                host_ip = item.status.addresses[0].address
+                host_name = item.status.addresses[1].address
+            break
+
+        # ssh to master node and check csi driver installed
+        command = "helm ls -n kube-system"
+        command_output = manager.get_command_output(host_name, command)
+        if len(command_output) == 1:
+            print("Driver not installed")
+            return
+
+        # ssh to master node and execute uninstall commands
+        command = "helm uninstall hpe-csi --namespace kube-system"
+        command_output = manager.get_command_output(host_name, command)
+        assert command_output[0] == 'release "hpe-csi" uninstalled', "Uninstall of 'hpe-csi' driver failed"
+        sleep(30)
+
+        # Delete crds
+        crd_obj = manager.hpe_list_crds()
+        for crd in crd_obj:
+            manager.hpe_delete_crd(crd)   
+            assert manager.check_if_deleted(timeout, crd, kind='Crd',namespace="Default") is True, \
+            "crd %s is not deleted yet " % crd
+
+        # Check plugin pods have been deleted
+        pod_obj = manager.hpe_list_pod_objects("kube-system")
+        pod_list = pod_obj.items
+        for item in pod_list:
+            if 'app' in item.metadata.labels:
+                if item.metadata.labels['app']=='hpe-csi-controller' or item.metadata.labels['app']=='hpe-csi-node' or item.metadata.labels['app']=='primera3par-csp':
+                    assert manager.check_if_deleted(timeout, item.metadata.name, "Pod", namespace=item.metadata.namespace) is True, \
+                    "Pod %s is not deleted yet " % pod.metadata.name     
+            else:
+                continue
+        print("Plugin pods deleted")        
+
+    except AssertionError as e:
+        raise e
+
+    except Exception as e:
+        raise e
+
+
+
+def test_helm_install():
+    try:
+        node_obj = manager.hpe_list_node_objects()
+        node_list = node_obj.items
+        # Fetching master node ip
+        for item in node_list:
+            if item.spec.taints != "None":
+                host_ip = item.status.addresses[0].address
+                host_name = item.status.addresses[1].address
+            break
+
+
+        # ssh to master node and check csi driver already installed
+        command = "helm ls -n kube-system"
+        command_output = manager.get_command_output(host_name, command)
+        if len(command_output) == 1:
+            print("Installing driver")
+        else:
+            print("Driver is already installed")
+            return
+
+
+        # copy values.yml file to /root dir on master node
+        cmd = "scp INSTALL/values_3par_1.18.yaml root@{}:/root".format(host_ip)
+        call(cmd.split(" ")) 
```

Comment:
Why are we copying file again, as discussed we do not need it right?

Reply:
We need to copy, as the code folder( git clone) and the landing path when we ssh to the master to execute the install command are different
clone path: ~/<clone dir>/INSTALL
master ssh path: /root

Keep it as scp, just incase in future this test is executed outside of master then a simple mv will fail. Therefore if we set the one time password less communication between hosts then scp is the way to copy between hosts.